Compiling with latest Xcode

I am now getting an error trying to build libzim on macOS using kiwix-build with my recently updated XCode.
AFAIK nothing else has changed

```
error: use of overloaded operator '!=' is ambiguous (with operand types 'std::reverse_iterator<zim::offset_t *>' and 'std::reverse_iterator<zim::offset_t *>')
```


```
[3/133] c++ -Itest/template.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/template.p/template.cpp.o -MF test/template.p/template.cpp.o.d -o test/template.p/template.cpp.o -c ../../SOURCE/libzim/test/template.cpp
[4/133] c++ -Itest/uuid.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/uuid.p/uuid.cpp.o -MF test/uuid.p/uuid.cpp.o.d -o test/uuid.p/uuid.cpp.o -c ../../SOURCE/libzim/test/uuid.cpp
[5/133] c++ -Itest/header.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/header.p/header.cpp.o -MF test/header.p/header.cpp.o.d -o test/header.p/header.cpp.o -c ../../SOURCE/libzim/test/header.cpp
[6/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/meson-generated_.._.._static_libzim-resources.cpp.o -MF src/libzim.8.dylib.p/meson-generated_.._.._static_libzim-resources.cpp.o.d -o src/libzim.8.dylib.p/meson-generated_.._.._static_libzim-resources.cpp.o -c static/libzim-resources.cpp
[7/133] c++ -Itest/cluster.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/cluster.p/tools.cpp.o -MF test/cluster.p/tools.cpp.o.d -o test/cluster.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[8/133] c++ -Itest/creator.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/creator.p/tools.cpp.o -MF test/creator.p/tools.cpp.o.d -o test/creator.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[9/133] c++ -Itest/error_in_creator.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/error_in_creator.p/tools.cpp.o -MF test/error_in_creator.p/tools.cpp.o.d -o test/error_in_creator.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[10/133] c++ -Itest/template.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/template.p/tools.cpp.o -MF test/template.p/tools.cpp.o.d -o test/template.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[11/133] c++ -Itest/dirent.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/dirent.p/tools.cpp.o -MF test/dirent.p/tools.cpp.o.d -o test/dirent.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[12/133] c++ -Itest/uuid.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/uuid.p/tools.cpp.o -MF test/uuid.p/tools.cpp.o.d -o test/uuid.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[13/133] c++ -Itest/lrucache.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/lrucache.p/tools.cpp.o -MF test/lrucache.p/tools.cpp.o.d -o test/lrucache.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[14/133] c++ -Itest/header.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/header.p/tools.cpp.o -MF test/header.p/tools.cpp.o.d -o test/header.p/tools.cpp.o -c ../../SOURCE/libzim/test/tools.cpp
[15/133] c++ -Itest/dirent.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/dirent.p/dirent.cpp.o -MF test/dirent.p/dirent.cpp.o.d -o test/dirent.p/dirent.cpp.o -c ../../SOURCE/libzim/test/dirent.cpp
[16/133] c++ -Itest/lrucache.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/lrucache.p/lrucache.cpp.o -MF test/lrucache.p/lrucache.cpp.o.d -o test/lrucache.p/lrucache.cpp.o -c ../../SOURCE/libzim/test/lrucache.cpp
../../SOURCE/libzim/test/lrucache.cpp:36:11: warning: unused variable 'NUM_OF_TEST1_RECORDS' [-Wunused-const-variable]
const int NUM_OF_TEST1_RECORDS = 100;
          ^
1 warning generated.
[17/133] c++ -Itest/cluster.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/cluster.p/cluster.cpp.o -MF test/cluster.p/cluster.cpp.o.d -o test/cluster.p/cluster.cpp.o -c ../../SOURCE/libzim/test/cluster.cpp
In file included from ../../SOURCE/libzim/test/cluster.cpp:53:
../../SOURCE/libzim/test/../src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[18/133] cc -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -std=c11 -O0 -g -MD -MQ src/libzim.8.dylib.p/md5.c.o -MF src/libzim.8.dylib.p/md5.c.o.d -o src/libzim.8.dylib.p/md5.c.o -c ../../SOURCE/libzim/src/md5.c
[19/133] c++ -Itest/creator.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/creator.p/creator.cpp.o -MF test/creator.p/creator.cpp.o.d -o test/creator.p/creator.cpp.o -c ../../SOURCE/libzim/test/creator.cpp
In file included from ../../SOURCE/libzim/test/creator.cpp:27:
../../SOURCE/libzim/test/../src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[20/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/buffer_reader.cpp.o -MF src/libzim.8.dylib.p/buffer_reader.cpp.o.d -o src/libzim.8.dylib.p/buffer_reader.cpp.o -c ../../SOURCE/libzim/src/buffer_reader.cpp
[21/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/cluster.cpp.o -MF src/libzim.8.dylib.p/cluster.cpp.o.d -o src/libzim.8.dylib.p/cluster.cpp.o -c ../../SOURCE/libzim/src/cluster.cpp
FAILED: src/libzim.8.dylib.p/cluster.cpp.o
c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/cluster.cpp.o -MF src/libzim.8.dylib.p/cluster.cpp.o.d -o src/libzim.8.dylib.p/cluster.cpp.o -c ../../SOURCE/libzim/src/cluster.cpp
In file included from ../../SOURCE/libzim/src/cluster.cpp:22:
In file included from ../../SOURCE/libzim/src/cluster.h:27:
In file included from ../../SOURCE/libzim/src/buffer.h:26:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/memory:858:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/ranges_uninitialized_algorithms.h:22:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__memory/uninitialized_algorithms.h:628:21: error: use of overloaded operator '!=' is ambiguous (with operand types 'std::reverse_iterator<zim::offset_t *>' and 'std::reverse_iterator<zim::offset_t *>')
    while (__first1 != __last1) {
           ~~~~~~~~ ^  ~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:914:27: note: in instantiation of function template specialization 'std::__uninitialized_allocator_move_if_noexcept<std::allocator<zim::offset_t>, std::reverse_iterator<zim::offset_t *>, std::reverse_iterator<zim::offset_t *>, zim::offset_t, void>' requested here
    __v.__begin_   = std::__uninitialized_allocator_move_if_noexcept(
                          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:1501:9: note: in instantiation of member function 'std::vector<zim::offset_t>::__swap_out_circular_buffer' requested here
        __swap_out_circular_buffer(__v);
        ^
../../SOURCE/libzim/src/cluster.cpp:113:19: note: in instantiation of member function 'std::vector<zim::offset_t>::reserve' requested here
    m_blobOffsets.reserve(n_offset);
                  ^
../../SOURCE/libzim/src/cluster.cpp:95:7: note: in instantiation of function template specialization 'zim::Cluster::read_header<unsigned long long>' requested here
      read_header<uint64_t>();
      ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__iterator/reverse_iterator.h:233:1: note: candidate function [with _Iter1 = zim::offset_t *, _Iter2 = zim::offset_t *]
operator!=(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
^
../../SOURCE/libzim/src/zim_types.h:96:34: note: candidate function [with T = std::reverse_iterator<zim::offset_t *>]
template<typename T> inline bool operator!=(const T& lhs, const T& rhs)
                                 ^
1 error generated.
[22/133] c++ -Itest/error_in_creator.p -Iinclude -I../../SOURCE/libzim/include -Isrc -I../../SOURCE/libzim/src -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -I/usr/local/Cellar/googletest/1.12.1_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DGTEST_HAS_PTHREAD=1 -DWITH_TEST_DATA=1 -MD -MQ test/error_in_creator.p/error_in_creator.cpp.o -MF test/error_in_creator.p/error_in_creator.cpp.o.d -o test/error_in_creator.p/error_in_creator.cpp.o -c ../../SOURCE/libzim/test/error_in_creator.cpp
../../SOURCE/libzim/test/error_in_creator.cpp:190:1: warning: 'InstantiateTestCase_P_IsDeprecated' is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
INSTANTIATE_TEST_CASE_P(
^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/gtest-param-test.h:503:38: note: expanded from macro 'INSTANTIATE_TEST_CASE_P'
  static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
                                     ^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/internal/gtest-internal.h:1295:1: note: 'InstantiateTestCase_P_IsDeprecated' has been explicitly marked deprecated here
GTEST_INTERNAL_DEPRECATED(
^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/internal/gtest-port.h:2289:59: note: expanded from macro 'GTEST_INTERNAL_DEPRECATED'
#define GTEST_INTERNAL_DEPRECATED(message) __attribute__((deprecated(message)))
                                                          ^
../../SOURCE/libzim/test/error_in_creator.cpp:360:1: warning: 'InstantiateTestCase_P_IsDeprecated' is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
INSTANTIATE_TEST_CASE_P(
^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/gtest-param-test.h:503:38: note: expanded from macro 'INSTANTIATE_TEST_CASE_P'
  static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
                                     ^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/internal/gtest-internal.h:1295:1: note: 'InstantiateTestCase_P_IsDeprecated' has been explicitly marked deprecated here
GTEST_INTERNAL_DEPRECATED(
^
/usr/local/Cellar/googletest/1.12.1_1/include/gtest/internal/gtest-port.h:2289:59: note: expanded from macro 'GTEST_INTERNAL_DEPRECATED'
#define GTEST_INTERNAL_DEPRECATED(message) __attribute__((deprecated(message)))
                                                          ^
2 warnings generated.
[23/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/dirent.cpp.o -MF src/libzim.8.dylib.p/dirent.cpp.o.d -o src/libzim.8.dylib.p/dirent.cpp.o -c ../../SOURCE/libzim/src/dirent.cpp
[24/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/envvalue.cpp.o -MF src/libzim.8.dylib.p/envvalue.cpp.o.d -o src/libzim.8.dylib.p/envvalue.cpp.o -c ../../SOURCE/libzim/src/envvalue.cpp
[25/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/fileheader.cpp.o -MF src/libzim.8.dylib.p/fileheader.cpp.o.d -o src/libzim.8.dylib.p/fileheader.cpp.o -c ../../SOURCE/libzim/src/fileheader.cpp
[26/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/blob.cpp.o -MF src/libzim.8.dylib.p/blob.cpp.o.d -o src/libzim.8.dylib.p/blob.cpp.o -c ../../SOURCE/libzim/src/blob.cpp
[27/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/buffer.cpp.o -MF src/libzim.8.dylib.p/buffer.cpp.o.d -o src/libzim.8.dylib.p/buffer.cpp.o -c ../../SOURCE/libzim/src/buffer.cpp
[28/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/template.cpp.o -MF src/libzim.8.dylib.p/template.cpp.o.d -o src/libzim.8.dylib.p/template.cpp.o -c ../../SOURCE/libzim/src/template.cpp
[29/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/dirent_accessor.cpp.o -MF src/libzim.8.dylib.p/dirent_accessor.cpp.o.d -o src/libzim.8.dylib.p/dirent_accessor.cpp.o -c ../../SOURCE/libzim/src/dirent_accessor.cpp
[30/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/archive.cpp.o -MF src/libzim.8.dylib.p/archive.cpp.o.d -o src/libzim.8.dylib.p/archive.cpp.o -c ../../SOURCE/libzim/src/archive.cpp
In file included from ../../SOURCE/libzim/src/archive.cpp:27:
In file included from ../../SOURCE/libzim/src/fileimpl.h:39:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[31/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/file_compound.cpp.o -MF src/libzim.8.dylib.p/file_compound.cpp.o.d -o src/libzim.8.dylib.p/file_compound.cpp.o -c ../../SOURCE/libzim/src/file_compound.cpp
In file included from ../../SOURCE/libzim/src/file_compound.cpp:21:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[32/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/file_reader.cpp.o -MF src/libzim.8.dylib.p/file_reader.cpp.o.d -o src/libzim.8.dylib.p/file_reader.cpp.o -c ../../SOURCE/libzim/src/file_reader.cpp
In file included from ../../SOURCE/libzim/src/file_reader.cpp:24:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[33/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/uuid.cpp.o -MF src/libzim.8.dylib.p/uuid.cpp.o.d -o src/libzim.8.dylib.p/uuid.cpp.o -c ../../SOURCE/libzim/src/uuid.cpp
[34/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/entry.cpp.o -MF src/libzim.8.dylib.p/entry.cpp.o.d -o src/libzim.8.dylib.p/entry.cpp.o -c ../../SOURCE/libzim/src/entry.cpp
In file included from ../../SOURCE/libzim/src/entry.cpp:25:
In file included from ../../SOURCE/libzim/src/fileimpl.h:39:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[35/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/item.cpp.o -MF src/libzim.8.dylib.p/item.cpp.o.d -o src/libzim.8.dylib.p/item.cpp.o -c ../../SOURCE/libzim/src/item.cpp
In file included from ../../SOURCE/libzim/src/item.cpp:25:
In file included from ../../SOURCE/libzim/src/fileimpl.h:39:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
1 warning generated.
[36/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/compression.cpp.o -MF src/libzim.8.dylib.p/compression.cpp.o.d -o src/libzim.8.dylib.p/compression.cpp.o -c ../../SOURCE/libzim/src/compression.cpp
[37/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/tools.cpp.o -MF src/libzim.8.dylib.p/tools.cpp.o.d -o src/libzim.8.dylib.p/tools.cpp.o -c ../../SOURCE/libzim/src/tools.cpp
[38/133] c++ -Isrc/libzim.8.dylib.p -Isrc -I../../SOURCE/libzim/src -Iinclude -I../../SOURCE/libzim/include -Istatic -I/Users/runner/./BUILD_native_dyn/INSTALL/include -I/Users/runner/BUILD_native_dyn/INSTALL/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -MD -MQ src/libzim.8.dylib.p/fileimpl.cpp.o -MF src/libzim.8.dylib.p/fileimpl.cpp.o.d -o src/libzim.8.dylib.p/fileimpl.cpp.o -c ../../SOURCE/libzim/src/fileimpl.cpp
In file included from ../../SOURCE/libzim/src/fileimpl.cpp:22:
In file included from ../../SOURCE/libzim/src/fileimpl.h:39:
../../SOURCE/libzim/src/file_compound.h:44:33: warning: 'binary_function<zim::Range, zim::Range, bool>' is deprecated [-Wdeprecated-declarations]
struct less_range : public std::binary_function< Range, Range, bool>
                                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__functional/binary_function.h:24:29: note: 'binary_function<zim::Range, zim::Range, bool>' has been explicitly marked deprecated here
struct _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 binary_function
                            ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:825:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
#    define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
                                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__config:810:49: note: expanded from macro '_LIBCPP_DEPRECATED'
#      define _LIBCPP_DEPRECATED __attribute__((deprecated))
                                                ^
../../SOURCE/libzim/src/fileimpl.cpp:384:21: warning: unused function 'direntCompareTitle' [-Wunused-function]
  static inline int direntCompareTitle(char ns, const std::string& title, const Dirent& dirent)
                    ^
2 warnings generated.
ninja: build stopped: subcommand failed.


Stopping build due to errors
Traceback (most recent call last):
  File "./kiwix-build/.github/scripts/build_release_nightly.py", line 56, in <module>
    run_kiwix_build(target, platform=PLATFORM_TARGET, make_release=MAKE_RELEASE)
  File "/Users/runner/kiwix-build/.github/scripts/common.py", line 188, in run_kiwix_build
    subprocess.check_call(command, cwd=str(HOME), env=env)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['kiwix-build', 'zim-tools', '--hide-progress', '--fast-clone', '--assume-packages-installed', '--target-platform', 'native_dyn']' returned non-zero exit status 1.
```

--- a/src/file_compound.h
+++ b/src/file_compound.h
@@ -41,7 +41,7 @@ struct Range {
   const offset_t max;
 };
 
-struct less_range : public std::binary_function< Range, Range, bool>
+struct less_range
 {
   bool operator()(const Range& lhs, const Range& rhs) const {
     return lhs.min < rhs.min && lhs.max <= rhs.min;

--- a/src/zim_types.h
+++ b/src/zim_types.h
@@ -31,74 +31,29 @@
 #define PACKED
 #endif
 
-
-template<typename B>
+template<typename B, typename S>
 struct REAL_TYPEDEF{
   typedef B base_type;
+  typedef S SELF;
+
   B v;
   REAL_TYPEDEF() : v(0) {};
   explicit REAL_TYPEDEF(B v) : v(v) {};
   explicit inline operator bool() const { return v != 0; }
   explicit inline operator B() const { return v; }
 
-  inline bool operator==(const REAL_TYPEDEF<B>& rhs) const
+  inline bool operator==(const REAL_TYPEDEF<B, S>& rhs) const
   { return v == rhs.v; }
 
-  inline REAL_TYPEDEF<B>& operator++()
+  inline REAL_TYPEDEF<B, S>& operator++()
   { v++; return *this; }
 
-  inline REAL_TYPEDEF<B> operator++(int)
-  { return REAL_TYPEDEF<B>(v++); }
+  inline REAL_TYPEDEF<B, S> operator++(int)
+  { return REAL_TYPEDEF<B, S>(v++); }
 } PACKED;
 
-template<typename T> inline T& operator+= (T& lhs, const T& rhs)
-{
-  lhs.v += rhs.v;
-  return lhs;
-}
-
-template<typename T> inline T& operator+= (T& lhs, const typename T::base_type& rhs)
-{
-  lhs.v += rhs;
-  return lhs;
-}
-
-template<typename T> inline T operator+(T lhs, const T& rhs)
-{
-  lhs += rhs;
-  return lhs;
-}
-
-template<typename T> inline T& operator-=(T& lhs, const T& rhs)
-{
-  lhs.v -= rhs.v;
-  return lhs;
-}
-
-template<typename T> inline T operator-(T lhs, const T& rhs)
-{
-  lhs -= rhs;
-  return lhs;
-}
-
-template<typename T> inline bool operator< (const T& lhs, const T& rhs)
-{ return lhs.v < rhs.v; }
-
-template<typename T> inline bool operator> (const T& lhs, const T& rhs)
-{ return rhs < lhs; }
-
-template<typename T> inline bool operator<=(const T& lhs, const T& rhs)
-{ return !(lhs > rhs); }
-
-template<typename T> inline bool operator>=(const T& lhs, const T& rhs)
-{ return !(lhs < rhs); }
-
-template<typename T> inline bool operator!=(const T& lhs, const T& rhs)
-{ return !(lhs == rhs); }
-
-
-template<typename B>
-std::ostream& operator<<(std::ostream& os, const REAL_TYPEDEF<B>& obj)
+template<typename B, typename S>
+std::ostream& operator<<(std::ostream& os, const REAL_TYPEDEF<B, S>& obj)
 {
     os << obj.v;
     return os;
@@ -106,9 +61,20 @@ std::ostream& operator<<(std::ostream& os, const REAL_TYPEDEF<B>& obj)
 
 namespace zim {
 
-#define TYPEDEF(NAME, TYPE) struct NAME : public REAL_TYPEDEF<TYPE> { \
-explicit NAME(TYPE v=0) : REAL_TYPEDEF<TYPE>(v) {}; } PACKED; \
-static_assert(sizeof(NAME) == sizeof(TYPE), "");
+#define TYPEDEF(NAME, TYPE) struct NAME : public REAL_TYPEDEF<TYPE, NAME> {          \
+explicit NAME(TYPE v=0) : REAL_TYPEDEF<TYPE, NAME>(v) {}; } PACKED;                  \
+static_assert(sizeof(NAME) == sizeof(TYPE), "");                                     \
+inline NAME& operator+= (NAME& lhs, const NAME& rhs) { lhs.v += rhs.v; return lhs; } \
+inline NAME& operator+= (NAME& lhs, const TYPE& rhs) { lhs.v += rhs; return lhs; }   \
+inline NAME operator+(NAME lhs, const NAME& rhs) { lhs += rhs; return lhs; }         \
+inline NAME& operator-=(NAME& lhs, const NAME& rhs) { lhs.v -= rhs.v; return lhs; }  \
+inline NAME operator-(NAME lhs, const NAME& rhs) { lhs -= rhs; return lhs; }         \
+inline bool operator< (const NAME& lhs, const NAME& rhs) { return lhs.v < rhs.v; }   \
+inline bool operator> (const NAME& lhs, const NAME& rhs) { return rhs < lhs; }       \
+inline bool operator<=(const NAME& lhs, const NAME& rhs) { return !(lhs > rhs); }    \
+inline bool operator>=(const NAME& lhs, const NAME& rhs) { return !(lhs < rhs); }    \
+inline bool operator!=(const NAME& lhs, const NAME& rhs) { return !(lhs == rhs); }
+
 
 TYPEDEF(entry_index_t, entry_index_type)
 TYPEDEF(title_index_t, entry_index_type)

--- a/test/lrucache.cpp
+++ b/test/lrucache.cpp
@@ -33,7 +33,6 @@
 #include "concurrent_cache.h"
 #include "gtest/gtest.h"
 
-const int NUM_OF_TEST1_RECORDS = 100;
 const int NUM_OF_TEST2_RECORDS = 100;
 const int TEST2_CACHE_CAPACITY = 50;
 

